### PR TITLE
Add functionality to remove users from teams

### PR DESCRIPTION
**Pull request by NotMyFault**

### CODE REVIEW

File: src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java [hunk 1]
```diff
@@ -737,6 +743,41 @@ private boolean addGitHubCommitter(Channel channel, User sender, String collabor
...
+            insufficientPermissionError(channel);
+            return false;
+        }
+        OutputChannel out = channel.send();
+        try {
+            if (justForThisRepo == null) {
+                // legacy command
+                out.message("I'm not longer managing the Everyone team. Please add committers to specific repos.");
+                return false;
+            }
```
Comment (timja) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java:758]: not needed in new commands / could be binned I think. or at least reword to something more sensible

File: src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java [hunk 2]
```diff
@@ -186,6 +186,12 @@ void handleDirectCommand(Channel channel, User sender, String payload) {
             return;
         }
 
+        m = Pattern.compile("(?:remove|revoke) (\\S+)(?: as)? (a )?(committ?er|member).*",CASE_INSENSITIVE).matcher(payload);
+        if (m.matches()) {
+            removeGitHubCommitter(channel,sender,m.group(1),null);
```
Comment (timja) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java:191]: the last arg is null so this will never work currently?

**Review by timja — COMMENTED** [review]
Needs docs PR to https://www.jenkins.io/projects/infrastructure/ircbot/

File: src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java (hunk 1 above)
Comment (NotMyFault) [reply src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java]: Removed it here, the IRC bot docs don't lose a word about this functionality, I guess it's time to move on and remove it here.

**Comment by NotMyFault** [comment]
> Needs docs PR to https://www.jenkins.io/projects/infrastructure/ircbot/

Is underway 👍🏻

File: src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java [hunk 3]
```diff
@@ -186,6 +186,12 @@ void handleDirectCommand(Channel channel, User sender, String payload) {
             return;
         }
 
+        m = Pattern.compile("(?:remove|revoke) (\\S+)(?: as)? (a )?(committ?er|member).*",CASE_INSENSITIVE).matcher(payload);
```
Comment (slide) [inline src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java:189]: I think you need something like the `on (.*)` here to get the `justForThisRepo` value.

Comment (NotMyFault) [reply src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java:189]: Right, reads weird otherwise, good catch 👍🏻

**Review by slide — APPROVED** [review]

**Comment by timja** [comment]
fine apart from docs missing

**Comment by NotMyFault** [comment]
Docs have been added and were just merged

**Pull request merged**